test/actions/katello/host/hypervisors_update_test.rb

```diff
@@ -59,6 +73,7 @@ class HypervisorsUpdateTest < ActiveSupport::TestCase
         @host = Host.find_by(:name => @hypervisor_name)
         assert_not_nil @host.subscription_facet
         assert_equal @facts['hypervisor.type'], @host.facts['hypervisor::type']
+        assert_equal @guests, @host.subscription_facet.virtual_guests.sort
```

Comment:
It's hard to tell by looking what is achieved by adding this assertion + the additional code you added to the setup method

An entirely new test in this file could be the solution (moving the setup code there too). Since there is no explicit test, someone could very easily come along and unknowingly remove the behavior which you are trying to test!

Reply:
Since this test file is to the the hypervisor update, my reason to add the extra codes to in the setup method is to make the data reusable across tests in the future. Minimum @guests and @hosts data are the basic data for the testing the hypervisor update I guess. The new assertion that I added is to strengthen the test case which is what we should check for the hypervisor update run. Maybe adding a comment explaining what the assertion trying to achieve will work?

Reply:
Looking at it now, I understand better. I agree with your approach - up to you if you want to add a comment about it, but it makes sense if you understand what the code is trying to do :)

Reply:
I ended up adding a new test. :D